ui/dashboards/src/components/DashboardToolbar/DashboardToolbar.tsx

```diff
@@ -30,26 +30,25 @@ export const DashboardToolbar = (props: DashboardToolbarProps) => {
   const { dashboardName } = props;
 
   const { isEditMode, setEditMode } = useEditMode();
-  const { openAddPanelGroup, openAddPanel, save } = useDashboardActions();
+  const { openAddPanelGroup, openAddPanel } = useDashboardActions();
   const isLaptopSize = useMediaQuery(useTheme().breakpoints.up('sm'));
-  const [originalSpec, setOriginalSpec] = useState<DashboardSpec | undefined>(undefined);
-  const { spec, resetSpec } = useDashboardSpec();
+  const [originalDashboard, setOriginalDashboard] = useState<DashboardResource | undefined>(undefined);
+  const { dashboardResource, resetDashboardResource } = useDashboardResource();
 
   const onEditButtonClick = () => {
-    setOriginalSpec(spec);
+    setOriginalDashboard(dashboardResource);
     setEditMode(true);
   };
 
   const onCancelButtonClick = () => {
     // Reset to the original spec and exit edit mode
-    if (originalSpec) {
-      resetSpec(originalSpec);
+    if (originalDashboard) {
+      resetDashboardResource(originalDashboard);
```

Comment:
I'm pumped for these changes, thank you for doing this.

Some overall thoughts:
- I'm not opposed to calling `useDashboard` (vs. `useDashboardResource`). I think the caller might appreciate the simplicity.
- I'm also not opposed to calling it `setDashboard`, (vs. `resetDashboard`), because I think that's essentially what it's doing? Taking whatever is given to it and setting it. 
- I think this has been planned already, but I think it would be cool if the management for the `originalDashboard` eventually moves into the provider/store, so that the caller doesn't have to do any of that. Then there could be functions like `reset` that internally take care of setting it back to the "original dashboard" (the dashboard at the time that edit mode begins).

Reply:
i was debating the two names too, and i think i like `useDashboard` better :)

> * I think it would be cool if the management for the `originalDashboard` eventually moves into the provider/store

the issue right now is that template variables and dashboard are two different stores, so essentially we will need another provider to wrap both template variables and dashboard stores if we don't want the caller to worry about `originalDashboard`